modules/base/rendering/renderablemodel.cpp

```diff
@@ -549,31 +653,68 @@ void RenderableModel::render(const RenderData& data, RendererTasks&) {
         case ColorAddingBlending:
             glBlendFunc(GL_SRC_COLOR, GL_DST_COLOR);
             break;
-    };
+        };
 
-    if (_disableDepthTest) {
-        glDisable(GL_DEPTH_TEST);
-    }
+        if (_disableDepthTest) {
+            glDisable(GL_DEPTH_TEST);
+        }
 
-    _geometry->render(*_program);
-    if (_disableFaceCulling) {
-        glEnable(GL_CULL_FACE);
-    }
+        _geometry->render(*_program);
+        if (_disableFaceCulling) {
+            glEnable(GL_CULL_FACE);
+        }
 
-    global::renderEngine->openglStateCache().resetBlendState();
+        global::renderEngine->openglStateCache().resetBlendState();
 
-    if (_disableDepthTest) {
-        glEnable(GL_DEPTH_TEST);
-    }
+        if (_disableDepthTest) {
+            glEnable(GL_DEPTH_TEST);
+        }
 
-    _program->deactivate();
+        _program->deactivate();
+    }
 }
 
-void RenderableModel::update(const UpdateData&) {
+void RenderableModel::update(const UpdateData& data) {
     if (_program->isDirty()) {
         _program->rebuildFromFile();
         ghoul::opengl::updateUniformLocations(*_program, _uniformCache, UniformNames);
     }
+
+    if (_geometry->hasAnimation() && _animationStart != "") {
+        double realtiveTime;
+        double now = data.time.j2000Seconds();
+        double startTime = data.time.convertTime(_animationStart);
+        double duration = _geometry->animationDuration();
+
+        switch (_animationMode) {
+            case AnimationMode::LoopFromStart:
```

Comment:
Maybe a comment here describing the idea behind remapping the relative time would be good for future-us that will come back to this code at some point